physics/checkpointer_body.hpp

```diff
@@ -64,55 +64,55 @@ Instant Checkpointer<Message>::checkpoint_at_or_before(Instant const& t) const {
 }
 
 template<typename Message>
-std::set<Instant> Checkpointer<Message>::all_checkpoints() const {
+absl::btree_set<Instant> Checkpointer<Message>::all_checkpoints() const {
   absl::ReaderMutexLock l(&lock_);
-  std::set<Instant> result;
+  absl::btree_set<Instant> result;
   std::transform(
       checkpoints_.cbegin(),
       checkpoints_.cend(),
       std::inserter(result, result.end()),
-      [](std::pair<Instant, typename Message::Checkpoint> const& pair) {
+      [](std::pair<Instant const, typename Message::Checkpoint> const& pair) {
```

Comment:
Interesting that this has an effect on performance.  I would favor giving a name to the map in the `.hpp` file thus:
```
using CheckpointsByTime = std::map<...>
```
and using `CheckpointsByTime::value_type` here.  That's one less thing to maintain.

Incidentally, should this map be a `btree_map`?

Reply:
Changed to `CheckpointsByTime::value_type`, and changed the actual type to `absl::btree_map`.
```
Benchmark                                                     Time             CPU      Time Old      Time New       CPU Old       CPU New
------------------------------------------------------------------------------------------------------------------------------------------
BM_CheckpointerOldestCheckpoint/1                          +0.0170         +0.0166            13            13            13            13
BM_CheckpointerOldestCheckpoint/8                          -0.0006         -0.0006            13            13            13            13
BM_CheckpointerOldestCheckpoint/64                         -0.0003         -0.0004            13            13            13            13
BM_CheckpointerOldestCheckpoint/512                        -0.0007         -0.0006            13            13            13            13
BM_CheckpointerNewestCheckpoint/1                          +0.0023         +0.0024            13            13            13            13
BM_CheckpointerNewestCheckpoint/8                          -0.0007         -0.0006            13            13            13            13
BM_CheckpointerNewestCheckpoint/64                         -0.0550         -0.0549            14            13            14            13
BM_CheckpointerNewestCheckpoint/512                        +0.0006         +0.0006            13            13            13            13
BM_CheckpointerCheckpointAtOrAfter/1                       +0.0014         +0.0015            13            13            13            13
BM_CheckpointerCheckpointAtOrAfter/8                       +0.0010         +0.0010            13            13            13            13
BM_CheckpointerCheckpointAtOrAfter/64                      +0.0402         +0.0403            13            14            13            14
BM_CheckpointerCheckpointAtOrAfter/512                     +0.1010         +0.1011            19            21            19            21
BM_CheckpointerCheckpointAtOrBefore/1                      +0.0036         +0.0035            13            13            13            13
BM_CheckpointerCheckpointAtOrBefore/8                      -0.0210         -0.0206            13            13            13            13
BM_CheckpointerCheckpointAtOrBefore/64                     -0.1688         -0.1684            16            13            16            13
BM_CheckpointerCheckpointAtOrBefore/512                    -0.1744         -0.1740            24            20            24            20
BM_CheckpointerAllCheckpoints/1                            -0.0250         -0.0248            82            80            82            80
BM_CheckpointerAllCheckpoints/8                            -0.0250         -0.0247           323           314           322           314
BM_CheckpointerAllCheckpoints/64                           -0.0484         -0.0480          1164          1108          1164          1108
BM_CheckpointerAllCheckpoints/512                          -0.1992         -0.1989          7114          5697          7111          5697
BM_CheckpointerAllCheckpointsAtOrBefore/1                  -0.0027         -0.0024            82            82            82            82
BM_CheckpointerAllCheckpointsAtOrBefore/8                  -0.0280         -0.0275           223           216           222           216
BM_CheckpointerAllCheckpointsAtOrBefore/64                 -0.0325         -0.0321           565           546           564           546
BM_CheckpointerAllCheckpointsAtOrBefore/512                -0.0310         -0.0307          2285          2214          2284          2214
BM_CheckpointerAllCheckpointsBetween/1                     -0.0286         -0.0282            13            13            13            13
BM_CheckpointerAllCheckpointsBetween/8                     -0.0105         -0.0101           226           223           225           223
BM_CheckpointerAllCheckpointsBetween/64                    -0.0184         -0.0179           562           551           561           551
BM_CheckpointerAllCheckpointsBetween/512                   -0.0423         -0.0418          2349          2250          2348          2250
```
`checkpoint_at_or_after` seems to actually be hurt by this change (albeit by only 2 ns). The other benchmarks seem to improve.